src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,1081 @@
+    {
+        uint firstId = _queue[client_].getNextId(LinkedIdList._SENTINEL);
+        if (firstId == LinkedIdList._SENTINEL) {
+            return false;
+        }
+
+        QueuedOrder storage order = _orders[client_][firstId];
+
+        // Skip if order is not in PENDING state.
+        if (order.state_ != RedemptionState.PENDING) {
+            _removeFromQueue(firstId, client_);
+            return false;
+        }
+
+        // Check token balance and allowance.
+        if (
+            !_validTokenBalance(
+                order.order_.paymentToken, order.client_, order.order_.amount
+            )
+        ) {
+            return false;
+        }
+
+        // Execute payment transfer
+        _executePaymentTransfer(firstId, order);
+        // return true as payment has been processed
+        return true;
+    }
+
+    /// @notice	Executes the actual payment transfer for an order. It sets
+    ///         the order state to PROCESSED if the transfer succeeds, otherwise
+    ///         if the transfer fails it sets the order state to FAILED. In both
+    ///         cases the order is removed from the queue.
+    /// @param	orderId_ The ID of the order to process.
+    /// @param	order_ The order to process.
+    function _executePaymentTransfer(uint orderId_, QueuedOrder memory order_)
+        internal
+        virtual
+    {
+        // Try to transfer payment from client to recipient
+        bool success_ = _tryPaymentTransfer(
+            order_.order_.paymentToken,
+            order_.client_,
+            order_.order_.recipient,
+            order_.order_.amount,
+            true // collect protocol fee when processing order
+        );
+
+        // Update order state based on transfer success
+        if (success_) {
+            _updateOrderState(
+                orderId_, address(order_.client_), RedemptionState.PROCESSED
+            );
+        } else {
+            _updateOrderState(
+                orderId_, address(order_.client_), RedemptionState.FAILED
+            );
+        }
+
+        // Remove processed order from queue
+        _removeFromQueue(orderId_, address(order_.client_));
+    }
+
+    /// @notice	This function does a low lever call to transfer
+    ///         funds to prevent reverts. Instead it will return a
+    ///         boolean value indicating whether the transfer was
+    ///         successful.
+    /// @param	token_ The token address.
+    /// @param	client_ The client address.
+    /// @param	recipient_ The recipient address.
+    /// @param	amount_ The amount to transfer.
+    /// @return	success_ True if the transfer was successful.
+    function _lowLevelTransfer(
+        address token_,
+        address client_,
+        address recipient_,
+        uint amount_
+    ) internal virtual returns (bool success_) {
+        // Make a low-level call to the token contract to execute transferFrom
+        (bool success, bytes memory data) = token_.call(
```

Comment:
A `try... catch.. ` pattern would do the same thing, but you do not need to parse the resuts of a low-level call yourself

Reply:
So it's also being tested with an address which doesn't have the transfer interface. I tried all kind of ways with `try/catch` but I didn't get it to work with the EVM revert which resulted from it. Eventually a colleague found this explanation for it not working, so I will stick with the low level call for now :)
![image (3)](https://github.com/user-attachments/assets/9717a384-5a86-4f69-b1f9-08315a861286)